pkg/aspect/metricsManager.go

```diff
@@ -0,0 +1,176 @@
+	dpb "istio.io/api/mixer/v1/config/descriptor"
+	"istio.io/mixer/pkg/adapter"
+	aconfig "istio.io/mixer/pkg/aspect/config"
+	"istio.io/mixer/pkg/attribute"
+	"istio.io/mixer/pkg/config"
+	"istio.io/mixer/pkg/expr"
+)
+
+type (
+	metricsManager struct {
+	}
+
+	metricInfo struct {
+		metricKind adapter.MetricKind
+		value      string
+		labels     map[string]string
+	}
+
+	metricsWrapper struct {
+		aspect   adapter.MetricsAspect
+		metadata map[string]*metricInfo // metric name -> info
+	}
+)
+
+// NewMetricsManager returns a manager for the metric aspect.
+func NewMetricsManager() Manager {
+	return &metricsManager{}
+}
+
+// NewAspect creates a metric aspect.
+func (m *metricsManager) NewAspect(c *config.Combined, a adapter.Builder, env adapter.Env) (Wrapper, error) {
+	params := c.Aspect.Params.(*aconfig.MetricsParams)
+
+	// TODO: get descriptors from config
+	desc := []*dpb.MetricDescriptor{
+		{
+			Name:  "api_responses",
+			Kind:  dpb.COUNTER,
+			Value: dpb.INT64,
+			Labels: []*dpb.LabelDescriptor{
+				{Name: "api_method", ValueType: dpb.STRING},
+				{Name: "response_code", ValueType: dpb.INT64},
+			},
+		},
+	}
+
+	metadata := make(map[string]*metricInfo)
+	defs := make([]adapter.MetricDefinition, len(desc))
+	for i, d := range desc {
+		def := toDefinition(d)
+		metric, found := find(params.Metrics, def.Name)
+		if !found {
+			env.Logger().Warningf("No metric found for descriptor %s, skipping it", def.Name)
+			continue
+		}
+
+		defs[i] = def
+		metadata[def.Name] = &metricInfo{
+			metricKind: def.Kind,
+			value:      metric.Value,
+			labels:     metric.Labels,
+		}
+	}
+
+	asp, err := a.(adapter.MetricsBuilder).NewMetricsAspect(env, c.Builder.Params.(adapter.AspectConfig), defs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct metrics aspect with config '%v' and err: %s", c, err)
+	}
+	return &metricsWrapper{asp, metadata}, nil
+}
+
+func (*metricsManager) Kind() string                        { return MetricKind }
+func (*metricsManager) DefaultConfig() adapter.AspectConfig { return &aconfig.MetricsParams{} }
+
+func (*metricsManager) ValidateConfig(adapter.AspectConfig) (ce *adapter.ConfigErrors) {
+	// TODO: we need to be provided the metric descriptors in addition to the metrics themselves here, so we can do type assertions.
+	// We also need some way to assert the type of the result of evaluating an expression, but we don't have any attributes or an
+	// evaluator on hand.
+	return
+}
+
+func (w *metricsWrapper) Execute(attrs attribute.Bag, mapper expr.Evaluator) (*Output, error) {
+	result := &multierror.Error{}
+	var values []adapter.Value
+
+metadataLoop:
+	for name, md := range w.metadata {
+		metricValue, err := mapper.Eval(md.value, attrs)
+		if err != nil {
+			result = multierror.Append(result, fmt.Errorf("failed to eval metric value for metric '%s' with err: %s", name, err))
+			continue metadataLoop // we can't satisfy this metric because we have the wrong types, so skip to the next
```

Comment:
do we need the label for the loop? 

it seems like, here, we could just use an unlabeled continue.

below, if we had a func that was `func mapLabels(expr.Evaluator, map[string]string) (map[string]interface{}, error)`, we could just have a simple:
```
labels, err := mapLabels(mapper, md.labels)
if err != nil {
    continue
}
```

That would be more readable, imho.

Even better if we added a method like: `func (*expr.Evaluator) EvalAll(map[string]string) (map[string]interface{}, error)`, so our code could be:
```
labels, err := mapper.EvalAll(md.labels)
if err != nil {
    continue
}
```